stl/inc/utility

```diff
@@ -644,6 +645,77 @@ template <size_t _Idx>
 inline constexpr in_place_index_t<_Idx> in_place_index{};
 #endif // _HAS_CXX17
 
+#if _HAS_CXX20
+
+template <typename _Ty>
+inline constexpr bool _Is_integral_not_bool_char = is_integral_v<_Ty>
+    && !is_same_v<_Ty, bool> && !is_same_v<_Ty, char>
+    && !is_same_v<_Ty, char8_t> && !is_same_v<_Ty, char16_t>
+    && !is_same_v<_Ty, char32_t> && !is_same_v<_Ty, wchar_t>;
+
+// SAFE INTEGRAL COMPARISONS
+template <typename _Ty1, typename _Ty2>
+constexpr bool in_range(const _Ty2 _Value) noexcept {
+    static_assert(_Is_integral_not_bool_char<_Ty1> && _Is_integral_not_bool_char<_Ty2>);
+    if constexpr (is_signed_v<_Ty1> && is_unsigned_v<_Ty2>) {
+        return _Value <= (numeric_limits<_Ty1>::max)();
+    } else if constexpr (is_unsigned_v<_Ty1> && is_signed_v<_Ty2>) {
+        return _Value >= 0 && _Value <= (numeric_limits<_Ty1>::max)();
+    } else {
+        return (numeric_limits<_Ty1>::min)() <= _Value && _Value <= (numeric_limits<_Ty1>::max)();
+    }
+}
+
+template <typename _Ty1, typename _Ty2>
+constexpr bool cmp_equal(const _Ty1 _Left, const _Ty2 _Right) noexcept {
+    static_assert(_Is_integral_not_bool_char<_Ty1> && _Is_integral_not_bool_char<_Ty2>);
+    if constexpr (is_signed_v<_Ty1> && is_unsigned_v<_Ty2>) {
+        return _Left >= 0 || static_cast<make_unsigned_t<_Ty1>>(_Left) == _Right;

```

Comment:
This seems to be incorrect. If _Left is >= 0 this is always true. So cmp_equal(1, 42) is true.
Replacing || with && is probably what you meant.

Reply:
Yes, it looks like an error. I'm still working on tests, so I’ll fix all the bugs soon.

Reply:
Thanks! fix in new commit.